tools/src/main/scala/scala/scalanative/codegen/CodeGen.scala

```diff
@@ -54,14 +57,53 @@ object CodeGen {
       // of available processesors. This prevents LLVM from optimizing
       // across IR module boundary unless LTO is turned on.
       def separate(): Seq[Path] =
-        partitionBy(assembly, procs)(_.name.top.mangle).par
+        assembly
+          .groupBy { defn =>
+            val packageName =
+              defn.name.top.id.split("\\.").dropRight(1).mkString(".")
```

Comment:
We can use single char split to make code simpler to read and not use regexes
```suggestion
              defn.name.top.id.split('.').init.mkString(".")
```